wgengine/netstack/netstack.go

```diff
@@ -891,7 +891,46 @@ func (ns *Impl) userPing(dstIP netip.Addr, pingResPkt []byte) {
 	switch runtime.GOOS {
 	case "windows":
 		err = exec.Command("ping", "-n", "1", "-w", "3000", dstIP.String()).Run()
-	case "darwin", "freebsd":
+	case "ios", "darwin":
+		_, payload := packet.ICMPEchoPayload(nil)
+
+		ifState := ns.dialer.NetMon().InterfaceState()
+		defaultIfIPs := ifState.InterfaceIPs[ifState.DefaultRouteInterface]
+		var srcIP netip.Addr
+		for _, pfx := range defaultIfIPs {
+			addr := pfx.Addr()
+			if dstIP.Is4() && addr.Is4() || dstIP.Is6() && addr.Is6() {
+				srcIP = addr
+				break
+			}
+		}
```

Comment:
maybe pull this out to a method on the netmon.State? Like `SrcIPForDstIP(dstIP) (_ netip.Addr, ok bool)` ?

Then handle `!ok`, which currently you don't and fall through to making a packet with a zero header?

Reply:
Went with a different approach (try to open a connection and check what the srcIP there is)